### ISSUE TITLE
How to get the program working?

### ISSUE THREAD

**Issue by kklepper** [iobroker-community-adapters/ioBroker.coronavirus-statistics#48]
**!!! Before you start !!!!**  
Verify if there is not already an issue requesting the same Enhancement

**Describe wanted Enhancement !**  
A clear description of the wanted functionality

Please give a step by step instruction. Your fine work is in vain if users cannot use it.

**Why should we put effort in it ?**  
Please add some additional information why this Enhancement should be integrated

I spent quite some time to find my way and switched to https://github.com/ahmadawais/corona-cli to get going, but I do not get there what I want (see https://github.com/ahmadawais/corona-cli/issues/94). I hope `corona-virus-statistics` answers some of the questions I have.

**Additional context**  

I am on windows (my linux box runs without a graphical interface), so I installed `iobroker-1.5.14.b-windows-installer.exe`. At first run, it showed a lot (guess: 50-100) adapters. I managed to install `corona-virus-statistics`, so I thought. 

Next I had to reboot my machine. ioBroker had only 3 adapters then. No idea where all the others had gone. I installed `corona-virus-statistics` again (alternative command `./iobroker url "https://github.com/iobroker-community-adapters/ioBroker.coronavirus-statistics/tarball/master" coronavirus-statistics`). It looked right and took some time, and when I looked again, I still had 3 adapters. No `corona-virus-statistics`. I could not believe it and repeated the procedure, with the same result.

Now what?

Obviously, your pals know what to do. I dont't and run out of ideas.

**Comment by DutchmanNL** [comment]
are you willing for an Webex session for some explanation of the ioBroker ?

Btw : did you read : https://www.iobroker.net/#de/documentation/admin/adapter.md

**Comment by kklepper** [comment]
> are you willing for an Webex session for some explanation of the ioBroker ?

Not yet.

> Btw : did you read : https://www.iobroker.net/#de/documentation/admin/adapter.md

Great advice! Thank you.

Now 

![image](https://user-images.githubusercontent.com/6130319/80728274-0c8fbd00-8b07-11ea-93d2-6c80bae83c07.png)

runs forever...

**Comment by DutchmanNL** [comment]
Anything in the Logs ?

**Comment by kklepper** [comment]
```

2020-04-30 17:16:19.669 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 terminated with code 1 ()
--
2020-04-30 17:23:25.277 - info: info.0 Popup-News readed...
2020-04-30 17:30:00.009 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 started with pid 22952
2020-04-30 17:30:00.016 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 started with pid 13112
2020-04-30 17:30:00.537 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 terminated with code 1 ()
2020-04-30 17:30:00.896 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 terminated with code 1 ()
2020-04-30 17:45:00.018 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 started with pid 20912
2020-04-30 17:45:00.026 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 started with pid 19916
2020-04-30 17:45:00.762 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 terminated with code 1 ()
2020-04-30 17:45:00.781 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 terminated with code 1 ()
2020-04-30 17:53:25.191 - info: info.0 Popup-News readed...
2020-04-30 18:00:00.019 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 started with pid 16356
2020-04-30 18:00:00.027 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 started with pid 3064
2020-04-30 18:00:00.533 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 terminated with code 1 ()
2020-04-30 18:00:00.534 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 terminated with code 1 ()
2020-04-30 18:15:00.008 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 started with pid 21556
2020-04-30 18:15:00.020 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 started with pid 5672
2020-04-30 18:15:00.383 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 terminated with code 1 ()
2020-04-30 18:15:00.388 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 terminated with code 1 ()
2020-04-30 18:23:25.203 - info: info.0 Popup-News readed...
2020-04-30 18:30:00.018 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 started with pid 23460
2020-04-30 18:30:00.175 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 started with pid 12024
2020-04-30 18:30:01.390 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 terminated with code 1 ()
2020-04-30 18:30:01.424 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 terminated with code 1 ()
2020-04-30 18:45:00.032 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 started with pid 23136
2020-04-30 18:45:00.228 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 started with pid 13104
2020-04-30 18:45:00.810 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 terminated with code 1 ()
2020-04-30 18:45:00.845 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 terminated with code 1 ()
2020-04-30 18:53:25.189 - info: info.0 Popup-News readed...
2020-04-30 19:00:00.008 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 started with pid 8484
2020-04-30 19:00:00.015 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 started with pid 12096
2020-04-30 19:00:00.611 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 terminated with code 1 ()
2020-04-30 19:00:00.617 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 terminated with code 1 ()
2020-04-30 19:15:00.019 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 started with pid 16400
2020-04-30 19:15:00.032 - info: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 started with pid 20324
2020-04-30 19:15:00.607 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.0 terminated with code 1 ()
2020-04-30 19:15:00.626 - error: host.w1412(SmartHome) instance system.adapter.coronavirus-statistics.1 terminated with code 1 ()
2020-04-30 19:23:25.229 - info: info.0 Popup-News readed...
etc. etc.
```

**Comment by DutchmanNL** [comment]
Hmm i dont see a reason here why the tables are not loading

As we had some start issue please do me a favor and delete the adapter + install again and see what happens

Otherwise I would really see pleas install teamviewer, anydesk or join me in a webex 😉

**Comment by kklepper** [comment]
Ok, will do ASAP

**Comment by kklepper** [comment]
I managed to uninstall `corona-virus-statistics`. 

Then I had learned to click on `Adapter mit Aktualisierungen` which had given me a long list and the opportunity to install `corona-virus-statistics` before. 

But not now. 

![image](https://user-images.githubusercontent.com/6130319/80752703-3dceb400-8b2c-11ea-97ce-a2f0ed7faf76.png)

I am stuck. Reload does not help. What now?

I had this before and used `Installieren aus eigener URL` and then `VON GITHUB`. This did something, but in the end I did not have the adapter, so it was in vain.

**Comment by kklepper** [comment]
I found a way via `Info`. `corona-virus-statistics` installs now.

**Comment by kklepper** [comment]
Stuck since 5 minutes:

![image](https://user-images.githubusercontent.com/6130319/80754426-4d033100-8b2f-11ea-979c-d91ae575f938.png)

Will resume...

**Comment by kklepper** [comment]
Same thing... starts with green progress button, progresses fast within 3 - 5 seconds, then turns red and is stuck.

**Comment by kklepper** [comment]
Started here:

![image](https://user-images.githubusercontent.com/6130319/80754693-bedb7a80-8b2f-11ea-974c-054f4c441ded.png)

**Comment by kklepper** [comment]
Similar with Admin except gets stuck and does not turn from green to red

![image](https://user-images.githubusercontent.com/6130319/80754861-0e21ab00-8b30-11ea-93e7-130494776eeb.png)

**Comment by kklepper** [comment]
Finally disappeared (while I wrote the protocol), but did not update...

![image](https://user-images.githubusercontent.com/6130319/80755012-550fa080-8b30-11ea-9407-1dd3da366372.png)

**Comment by kklepper** [comment]
Same with discovery

![image](https://user-images.githubusercontent.com/6130319/80755116-8e481080-8b30-11ea-83b6-c310787767c1.png)

I notice that node.js is outdated... and catch the message that update had been successfull... So I was wrong. 

Refresh shows new status:

![image](https://user-images.githubusercontent.com/6130319/80755309-dcf5aa80-8b30-11ea-89a0-959cd4b63e37.png)

`Info` is up-to-date as well

![image](https://user-images.githubusercontent.com/6130319/80755451-1fb78280-8b31-11ea-8fe0-7c4cdc2845e6.png)


Ok, new attempt...

**Comment by kklepper** [comment]
Here we have it:

![image](https://user-images.githubusercontent.com/6130319/80755803-bedc7a00-8b31-11ea-9885-a18479c60d52.png)

$ ./iobroker install coronavirus-statistics
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
adapter "coronavirus-statistics" already installed. Use "upgrade" to upgrade to a newer version.
ERROR: process exited with code 51

**Comment by kklepper** [comment]
`adapter "coronavirus-statistics" already installed. Use "upgrade" to upgrade to a newer version.`

I could not find the adapter so I used `Alle Adapter upgraden`

![image](https://user-images.githubusercontent.com/6130319/80756086-390cfe80-8b32-11ea-9e20-8bda613c0fde.png)

Well, I am not convinced. I just do not see the adapter. Conjecture: my manual install might have been wrong. 

So 

```D:\node_modules>npm uninstall iobroker.coronavirus-statistics@latest
npm WARN saveError ENOENT: no such file or directory, open 'D:\package.json'
npm WARN enoent ENOENT: no such file or directory, open 'D:\package.json'
npm WARN !invalid#1 No description
npm WARN !invalid#1 No repository field.
npm WARN !invalid#1 No README data
npm WARN !invalid#1 No license field.

audited 315 packages in 7.404s
found 5 low severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details```

Next a new attempt:

```$ ./iobroker install coronavirus-statistics
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
Socket.io Server detected. Please update to js-controller 2.0 or higher!
adapter "coronavirus-statistics" already installed. Use "upgrade" to upgrade to a newer version.
ERROR: process exited with code 51```

**Comment by kklepper** [comment]
`Socket.io Server detected. Please update to js-controller 2.0 or higher!` 

I'll tackle this one now.

**Comment by DutchmanNL** [comment]
wow JS-Controller < 2.0, which installer did u use that's basically almost Stone Age (mja a year ago but within it that's Stone Age :))

**Comment by kklepper** [comment]
Where do I find `JS-Controller`? I assumed it is part of node.js and updated to the latest version 

```
D:\>node -v
v12.16.3
```

but  ioBroker after refresh still shows 

```
Node.js
v10.16.3 (Es gibt eine neuere Version: v10.20.1 - Empfohlene Version v12.16.3)
```

Now the additional packages are being installed (right now python) and after that I will reboot because of Windows update. Hopefully this will fix the problem.

**Comment by DutchmanNL** [comment]
> Where do I find `JS-Controller`? I assumed it is part of node.js and updated to the latest version

no JS-controller is the MASTER of iobroker, the brain, the backend .... the hart of everything

as I see your setup is really wrong, were did you install ioBroker from this is going the wrong direction :) ?

**Comment by kklepper** [comment]
I used the latest download 2020-04-29: `iobroker-1.5.14.b-windows-installer.exe`. What is wrong with this?

**Comment by kklepper** [comment]
Same thing:

```
Node.js
v10.16.3 (Es gibt eine neuere Version: v10.20.1 - Empfohlene Version v12.16.3)
```

How come?

**Comment by kklepper** [comment]
Ok, found `iobroker-latest-windows-installer` , try again.

**Comment by kklepper** [comment]
Now this is the new installation running on a new port:

![image](https://user-images.githubusercontent.com/6130319/80807951-baaa6e00-8bbe-11ea-99e5-6a8001ae6db7.png)

**Comment by kklepper** [comment]
> js-controller Upgrade Anleitung
> 
> Für das Updaten von ioBroker auf Windows den entsprechenden Installer mit der gewünschten js-controller-Version von der Download-Seite https://www.iobroker.net/#de/download herunterladen und mit diesem das Update vornehmen. Mit dem Windows Installer können auch vorher manuell installierte Server oder Installationen von anderen Betriebssystemen nach Windows migriert und geupdated werden.
> 

Did that, result known.

> Windows (manuell installiert)
> Eine manuelle Installation erfolgt mit Administrator-Rechten. Bitte deshalb ein cmd.exe-Kommandozeilenfenster als Administrator starten (mit Rechtsklick auf cmd.exe und "Ausführen als Administrator") und dort folgende Befehle ausführen:
> 
> cd C:\iobroker (oder wo ioBroker installiert wurde)
> iobroker stop, um den ioBroker Service zu stoppen
> iobroker status, um zu prüfen, ob ioBroker beendet ist
> iobroker update
> iobroker upgrade self
> ioBroker Service starten oder Rechner rebooten, danach sollte ioBroker neu starten und man kann sicher sein das alle alten Prozesse beendet waren.

Does not work...

```
D:\>"D:\data\Programme\iobroker\ioBroker\daemon\iobroker" stop
2020-05-01 15:28:19,874 INFO  - Stopping the service with id 'iobroker(ioBroker)'

D:\>"D:\data\Programme\iobroker\ioBroker\daemon\iobroker" status
Stopped

D:\>"D:\data\Programme\iobroker\ioBroker\daemon\iobroker" update
Unknown command: update
Available commands:
- 'install'   - install the service to Windows Service Controller
- 'uninstall' - uninstall the service
- 'start'     - start the service (must be installed before)
- 'stop'      - stop the service
- 'restart'   - restart the service
- 'restart!'  - self-restart (can be called from child processes)
- 'status'    - check the current status of the service
- 'test'      - check if the service can be started and then stopped
- 'testwait'  - starts the service and waits until a key is pressed then stops the service
- 'version'   - print the version info
- 'help'      - print the help info (aliases: -h,--help,-?,/?)
2020-05-01 15:41:12,436 FATAL - Unhandled exception
System.Exception: Unknown command: update
   bei winsw.WrapperService.Run(String[] _args, ServiceDescriptor descriptor)
   bei winsw.WrapperService.Main(String[] args)
System.Exception: Unknown command: update
   bei winsw.WrapperService.Run(String[] _args, ServiceDescriptor descriptor)
   bei winsw.WrapperService.Main(String[] args)

D:\>"D:\data\Programme\iobroker\ioBroker\daemon\iobroker" upgrade self
Unknown command: upgrade
Available commands:
- 'install'   - install the service to Windows Service Controller
- 'uninstall' - uninstall the service
- 'start'     - start the service (must be installed before)
- 'stop'      - stop the service
- 'restart'   - restart the service
- 'restart!'  - self-restart (can be called from child processes)
- 'status'    - check the current status of the service
- 'test'      - check if the service can be started and then stopped
- 'testwait'  - starts the service and waits until a key is pressed then stops the service
- 'version'   - print the version info
- 'help'      - print the help info (aliases: -h,--help,-?,/?)
2020-05-01 15:50:54,387 FATAL - Unhandled exception
System.Exception: Unknown command: upgrade
   bei winsw.WrapperService.Run(String[] _args, ServiceDescriptor descriptor)
   bei winsw.WrapperService.Main(String[] args)
System.Exception: Unknown command: upgrade
   bei winsw.WrapperService.Run(String[] _args, ServiceDescriptor descriptor)
   bei winsw.WrapperService.Main(String[] args)
   
D:\>"D:\data\Programme\iobroker\ioBroker\daemon\iobroker" version
WinSW 2.3.0.0

D:\>"D:\data\Programme\iobroker\ioBroker\daemon\iobroker" start
2020-05-01 15:41:51,129 INFO  - Starting the service with id 'iobroker(ioBroker)'

D:\>"D:\data\Programme\iobroker\ioBroker\daemon\iobroker" status
Started

```

**Comment by Stabilostick** [comment]
Please change your your active folder one level up before issuing the update command. iobroker.exe in the daemon folder is only a service wrapper, not the main iobroker program.

**Comment by kklepper** [comment]
This is really weird:

```
D:\>node -v
v12.16.3

```

But 

```
Node.js
v10.17.0 (Es gibt eine neuere Version: v10.20.1 - Empfohlene Version v12.16.3)
```

Preinstalled adapters outdated again. Refresh hangs

![image](https://user-images.githubusercontent.com/6130319/80810940-2abbf280-8bc5-11ea-850e-66869d9102d4.png)

Same with discovery

![image](https://user-images.githubusercontent.com/6130319/80811524-5ab7c580-8bc6-11ea-8972-7435d61c9984.png)

Is the effort worth it? 

Should I report to https://forum.iobroker.net/category/10/error-bug ?

Once again: 

- Adapter - all were shown
- discovery refreshed, refresh admin went like a breeze

Next 

![image](https://user-images.githubusercontent.com/6130319/80813297-14646580-8bca-11ea-9ba7-696de60e8c19.png)

for 15 minutes, no log entries

**Comment by kklepper** [comment]
> Please change your active folder one level up before issuing the update command. iobroker.exe in the daemon folder is only a service wrapper, not the main iobroker program.

Ok, will do...

**Comment by kklepper** [comment]
![image](https://user-images.githubusercontent.com/6130319/80813471-61483c00-8bca-11ea-9c57-78cf8adc2438.png)

**Comment by kklepper** [comment]
@Stabilostick

```
D:\>cd "D:\data\Programme\iobroker\ioBroker\daemon\iobroker"
Das System kann den angegebenen Pfad nicht finden.

D:\>cd "D:\data\Programme\iobroker\ioBroker\daemon\"

D:\data\Programme\iobroker\ioBroker\daemon>cd ..

D:\data\Programme\iobroker\ioBroker>iobroker status
*** Error. Please start the ioBroker commandline with the correct Windows startmenu shortcut. Aborting...

```

Ok, do that...

```
**********************************************************
*               Welcome to ioBroker.                 *
*                                                    *
*   Type 'iobroker help' for list of instructions.   *
*                For more help see                   *
*     https://github.com/ioBroker/ioBroker.docs      *
**********************************************************

Your environment has been set up for using Node.js 10.17.0 (x64) and npm.
```
Looks good so far... `update` looks fine, long list...

```
D:\data\Programme\iobroker\ioBroker>iobroker upgrade self
Update js-controller from @2.1.0 to @2.2.9
NPM version: 6.11.3
npm install iobroker.js-controller@2.2.9 --unsafe-perm --loglevel error --prefix "D:/data/Programme/iobroker/ioBroker" (System call)
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (D:\data\Programme\iobroker\ioBroker\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:193:23)
gyp ERR! stack     at ChildProcess.emit (events.js:198:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Windows_NT 10.0.18363
gyp ERR! command "D:\\data\\Programme\\iobroker\\ioBroker\\nodejs\\node.exe" "D:\\data\\Programme\\iobroker\\ioBroker\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd D:\data\Programme\iobroker\ioBroker\node_modules\iobroker.js-controller\node_modules\unix-dgram
gyp ERR! node -v v10.17.0
gyp ERR! node-gyp -v v5.0.3
gyp ERR! not ok
Starting node restart.js
```

**Comment by kklepper** [comment]
I added 5 SQL results with same query, different order to https://github.com/ahmadawais/corona-cli/issues/94

**Comment by Stabilostick** [comment]
Okay, I assume the update went well?

**Comment by kklepper** [comment]
@Stabilostick see above `gyp ERR! not ok`

**Comment by Stabilostick** [comment]
Unix-dgram can not be build on Windows OS. But this is not a problem because it is an optional package and not needed on Windows.

**Comment by kklepper** [comment]
@Stabilostick Thank you! So it should work, right?

The issue with js-controller seems to be resolved.

Issues with `http://localhost:8091/adapter/info/tab_m.html#`

- `Meine Adapter` does not stop.
- Node.js
v10.17.0 (Es gibt eine neuere Version: v10.20.1 - Empfohlene Version v12.16.3)

Remember: 
```
D:\node_modules>node -v
v12.16.3

```
![image](https://user-images.githubusercontent.com/6130319/80873322-f117e400-8cb7-11ea-9026-c5e374bdb746.png)

Adapter `coronavirus-statistics` could be installed, but

`http://localhost:8091/#tab-instances/config/system.adapter.coronavirus-statistics.1` does not stop as before

![image](https://user-images.githubusercontent.com/6130319/80873423-b8c4d580-8cb8-11ea-8a98-0d78040c06e5.png)

Klicking on `ERWEITERTE EINSTELLUNGEN` stopped this.

Now I have it twice:

![image](https://user-images.githubusercontent.com/6130319/80873513-5ae4bd80-8cb9-11ea-8e20-4e79fda95977.png)

Ok, interesting, I expected one to be there from yesterday, but did not find it. So you can have many adapters of the same type.

- Does cron work on Windows? 
- How do I get information now from an adapter?

Ok, I'll look for information via Google first.

**Comment by kklepper** [comment]
I watched installation on pi and Tankstellenpreise, so I have learned something and resume:

- I should install `ioBroker` on my Linux box and work from Windows
- from Adapter `coronavirus-statistics` I probably only get raw data
- working out the results I got with SQL is some work 

Correct?

Anyway, a new whole world I knew nothing about opened up for me.

**Comment by kklepper** [comment]
I learned about sql and flot and vis and expected to find data under `coronavirus-statistics` objects -- but I only find `countryTranslator`.

So what now? I developed the idea to feed the official numbers as datapoints to be able to generate charts, but where do I find data?

**Comment by kklepper** [comment]
FYI:
https://coronavirus-19-api.herokuapp.com/tabs/tab1
Information from www.worldometers.info

=> see https://github.com/CSSEGISandData/COVID-19/issues/2417

**Comment by DutchmanNL** [comment]
I am not sure without anydesk or team viewer how to assist further

**Comment by kklepper** [comment]
Ok, I try with Linux. Thank you all.

**Comment by kklepper** [comment]
Tested 2 docker images

```
docker@boot2docker:/mnt/sde1/tmp$ docker images | grep "iobroker"
buanet/iobroker                                                          latest                  cde9c9ca1a33        5 days ago          1GB
salenss/iobroker                                                         latest                  34214e6f9c92        16 months ago       405MB
```
buanet/iobroker worked, will have a close look now and try to generate graphs.

**Comment by kklepper** [comment]
With buanet/iobroker, I installed the adapter and see all the objects. Fine.

As far as I can see there are `coronavirus-statistics` widgets:

```
root@bec1c88cf301:/opt/iobroker#  grep -rn 'iobroker-data' -e "ioBroker.vis coronavirus-statistics Widget-Set"
iobroker-data/files/vis/index.html:3260:    ioBroker.vis coronavirus-statistics Widget-Set
iobroker-data/files/vis/edit.html:3281:    ioBroker.vis coronavirus-statistics Widget-Set
iobroker-data/files/vis/widgets/coronavirus-statistics.html:2:    ioBroker.vis coronavirus-statistics Widget-Set
```

I also have `http://voxx.b2d:8082/vis/edit.html#corona1`  and inserted an `iframe` and `html` widget. 

I tried several methods (path, direct paste, Widgets Importieren), nothing worked.

For example the latter produces the same error regardless if path (`iobroker-data/files/vis/index.html`) or code (`https://raw.githubusercontent.com/iobroker-community-adapters/ioBroker.coronavirus-statistics/master/widgets/coronavirus-statistics.html`) is inserted:

![image](https://user-images.githubusercontent.com/6130319/80926632-b5a11680-8d98-11ea-95e6-6dba37be15e5.png)

From this I deduce that widgets have to be `json`, not `html`.

For the `iframe`: what is the path? I tired to find out and used numerous variants, to no avail. Sorry, this is what I meant:

> Please give a step by step instruction. Your fine work is in vain if users cannot use it.

I spent hours surfing the internet and studying different tutorials, and I still don't get it.

How do I get your widgets to be shown?
